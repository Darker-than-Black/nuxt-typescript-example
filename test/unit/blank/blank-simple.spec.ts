import VueTestInstance from '~/test/factory/vue-instance'
import BlankSimple from '~/components/blank/items/blank-simple.vue'
import DECLARATION_BLANK_ITEMS from '~/assets/js/glossary/balnk/declaration-blank-items'
import DeclarationMock from '~/test/api/declaration-info.json'

describe('blank-simple.vue', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = VueTestInstance(BlankSimple, {
      options: {
        propsData: {
          config: DECLARATION_BLANK_ITEMS[1],
          data: DeclarationMock
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
