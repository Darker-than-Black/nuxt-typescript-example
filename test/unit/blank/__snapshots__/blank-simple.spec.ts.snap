// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blank-simple.vue should match the snapshot 1`] = `
<section class="blank-item">
  <div class="blank-item__header">
    <div>
      <div class="blank-item__gray">
        1. Тип декларації
      </div>
      <ul class="blank-item__list">
        <li title="—" class="blank-item__text blank-item__btn fixed-length">
          —
        </li>
        <li title="—" class="blank-item__text blank-item__btn fixed-length">
          —
        </li>
        <li title="—" class="blank-item__text blank-item__btn fixed-length">
          —
        </li>
      </ul>
    </div>
  </div>
  <div class="blank-item__content grid">
    <div class="blank-item__content--element w-50">
      <div class="blank-item__gray">
        3. Додаткові аркуші
      </div>
      <div class="blank-item__text blank-item__flex">
        —
      </div>
    </div>
    <div class="blank-item__content--element w-50">
      <div class="blank-item__gray">
        4. Відвант. спец.
      </div>
      <div class="blank-item__text blank-item__flex">
        —
      </div>
    </div>
    <div class="blank-item__content--element w-50">
      <div class="blank-item__gray">
        5. Товарів
      </div>
      <div class="blank-item__text blank-item__flex">
        —
      </div>
    </div>
    <div class="blank-item__content--element w-50">
      <div class="blank-item__gray">
        6. Місць
      </div>
      <div class="blank-item__text blank-item__flex">
        —
      </div>
    </div>
  </div>
</section>
`;
